src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -346,10 +349,23 @@ object ApiOperations extends DataOperations with MetadataOperations {
       predicate.copy(set = predicate.set.map(field => sanitizeForSql(field.toString)))
     }
 
+  /** Sanitizes aggregation format so query is safe from SQL injection */
+  def sanitizeAggregations(fields: List[Field]): List[Field] =
+    fields.map {
+      case sf: SimpleField => sf
+      case FormattedField(field, function, format) =>
+        FormattedField(sanitizeForSql(field), function, sanitizeDatePartAggregation(format))
+    }
+
   /** Sanitizes string to be viable to paste into plain SQL */
   def sanitizeForSql(str: String): String = {
     val supportedCharacters = Set('_', '.', '+', ':', '-', ' ')
     str.filter(c => c.isLetterOrDigit || supportedCharacters.contains(c))
   }
 
+  /** Sanitizes datePart aggregate function*/
+  def sanitizeDatePartAggregation(str: String): String = {
+    val supportedCharacters = Set('Y', 'M', 'D', 'A', '-')
```

Comment:
This doesn't seem to line up with the standard [Java date format](https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html).

Reply:
way I did is described here: https://rajivrnair.dev/postgres-grouping, it's not java date format, it's postgres